ocean_provider/validation/RBAC.py

```diff
@@ -77,6 +77,41 @@ def get_additional_dids(self):
             for additional_input in additional_inputs
         ]
 
+    def get_data(self):
+        if "data" not in self.request.keys():
+            raise Exception("Data to encrypt is empty.")
+        if not self._is_asset() and not self._is_file_encryption_data():
+            raise Exception("Invalid type of data.")
+
+        return self.request["data"]
+
+    def _is_asset(self) -> bool:
+        data = self.request["data"]
+        if not isinstance(data, dict) and not isinstance(data, str):
+            return False
+
+        if isinstance(data, str):
+            try:
+                data = json.loads(data)
+            except:
+                return False
+
+        if isinstance(data, dict):
+            return data.get("version", False)
+
+    def _is_file_encryption_data(self) -> bool:
+        data = self.request["data"]
+        if not isinstance(data, list):
+            return False
+
+        for file in data:
+            if (
+                isinstance(file, dict)
+                and list(file.keys()) == ["nftAddress", "datatokenAddress", "files"]
+                and isinstance(file["files"], dict)
+            ):
+                return True
```

Comment:
shouldn't this be higher-level? Is it a file structure if just only one of the elements in data has the correct structure?

can you also look into the list comparison possible issues? What if I add the nftAddress, datatokenAddress and files in a different order? Will comparing with `==` work?

can files also be stringified?

I think we should only handle the data encryption separately, and leave the rest as is, without restrictions. As far as I know, anything should be encryptable.

Reply:
And actually, people can create their own private aquariuses with different schemas, so we shouldn't do any validation on this structure.